sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/service/schema/JsonSchemaManagerTest.java

```diff
@@ -0,0 +1,188 @@
+            .collect(Collectors.toSet());
+
+    Schema.CqlTable.Builder table =
+        Schema.CqlTable.newBuilder()
+            .setName(collection)
+            .putOptions("comment", "{\"schema\": {}}")
+            .addPartitionKeyColumns(partitionColumn)
+            .addAllClusteringKeyColumns(clusteringColumns)
+            .addAllColumns(valueColumns);
+    return Schema.CqlKeyspaceDescribe.newBuilder()
+        .setCqlKeyspace(Schema.CqlKeyspace.newBuilder().setName(namespace))
+        .addTables(table)
+        .build();
+  }
+
+  @Nested
+  class AttachJsonSchema {
+    @Test
+    public void happyPath() throws JsonProcessingException {
+      String namespace = RandomStringUtils.randomAlphanumeric(16);
+      String collection = RandomStringUtils.randomAlphanumeric(16);
+
+      QueryOuterClass.Response response = QueryOuterClass.Response.newBuilder().build();
+
+      doAnswer(
+              invocationOnMock -> {
+                StreamObserver<QueryOuterClass.Response> observer = invocationOnMock.getArgument(1);
+                observer.onNext(response);
+                observer.onCompleted();
+                return null;
+              })
+          .when(bridgeService)
+          .executeQuery(any(), any());
+
+      JsonNode schema =
+          objectMapper.readTree("{\"$schema\": \"https://json-schema.org/draft/2019-09/schema\"}");
+      jsonSchemaManager
+          .attachJsonSchema(namespace, collection, schema)
+          .subscribe()
+          .withSubscriber(UniAssertSubscriber.create())
+          .awaitItem();
+
+      verify(bridgeService).executeQuery(any(), any());
+      verifyNoMoreInteractions(bridgeService);
+    }
+  }
+
+  @Nested
+  class GetJsonSchema {
+
+    @Test
+    public void happyPath() throws JsonProcessingException {
+      String namespace = RandomStringUtils.randomAlphanumeric(16);
+      String collection = RandomStringUtils.randomAlphanumeric(16);
+
+      Schema.CqlKeyspaceDescribe keyspace = getValidTableAndKeyspace(namespace, collection);
+
+      doAnswer(
+              invocationOnMock -> {
+                StreamObserver<Schema.CqlKeyspaceDescribe> observer =
+                    invocationOnMock.getArgument(1);
+                observer.onNext(keyspace);
+                observer.onCompleted();
+                return null;
+              })
+          .when(bridgeService)
+          .describeKeyspace(any(), any());
+
+      UniAssertSubscriber<JsonNode> result =
+          jsonSchemaManager
+              .getJsonSchema(tableManager, namespace, collection)
+              .subscribe()
+              .withSubscriber(UniAssertSubscriber.create());
+
+      result
+          .awaitItem()
+          .assertItem(objectMapper.readTree(keyspace.getTables(0).getOptionsMap().get("comment")))
+          .assertCompleted();
+
+      verify(bridgeService).describeKeyspace(any(), any());
+      verifyNoMoreInteractions(bridgeService);
+    }
+  }
+
+  @Nested
+  class ValidateJsonSchema {
```

Comment:
missing cases:

* no schema in table
* schema validation fails (object not complied with schema)